### PR TITLE
Add msys recipe

### PR DESCRIPTION
**Pull request by rmcgibbo**
Work in progress recipe for msys (https://github.com/DEShawResearch/msys)

### CODE REVIEW

**Comment by isuruf** [comment]
We might have to rename the package. Even though there is no package named msys, we mean http://www.mingw.org/wiki/msys or https://www.msys2.org/ by the name msys.

cc @conda-forge/core, @mingwandroid

File: recipes/msys/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{% set name = "msys" %}
+{% set version = "1.7.258" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - git_url: git://github.com/rmcgibbo/msys.git
+    git_rev: "cf53228"
```
Comment (scopatz) [inline recipes/msys/meta.yaml]: This doesn't seem right: https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos

File: recipes/msys/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - scons
+    - python
+    - numpy
+    - boost>=1.67.0
```
Comment (scopatz) [inline recipes/msys/meta.yaml]: There should be a space after the `boost`, ie. `boost >=1.67.0`.  Also, are you sure you want boost and not boost-cpp?

**Comment by mingwandroid** [comment]
I personally don't care about this name clash. MSys might though. Is there still people on that project, I don't know.

File: recipes/msys/meta.yaml (hunk 2 above)
Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: I'll give that a try

File: recipes/msys/meta.yaml (hunk 1 above)
Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: Yeah, as I said in the first update of the PR, this is still a work-in-progress. I've been iterating on the base recipe in order to debug build failures. When this is ready to go, I definitely won't be trying to build from git.

Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: I'd obviously like to merge the changes I've made to the project's SConscript upstream to https://github.com/DEShawResearch/msys before we merge this packaging PR.

File: recipes/msys/meta.yaml (hunk 2 above)
Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: I do want boost, rather than just boost-cpp (boost/python.hpp)

File: recipes/msys/meta.yaml (hunk 1 above)
Comment (CJ-Wright) [reply recipes/msys/meta.yaml]: You could also apply those changes as a patch.

Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: The lead upstream developer is in the office two doors down from me, so I think we'll get it merged upstream shortly. :)

Do ya'll prefer building from a "release" tarball on github, or a tarball uploaded elsewhere?

Comment (CJ-Wright) [reply recipes/msys/meta.yaml]: It is your choice. We have a bot which will issue PRs for version bumps if the tarballs are on most major platforms, including GitHub and PyPI. The bot will also try other places (eg. sourceforge), but doesn't always find the tarballs.

Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: Building against the tarball now.

**Comment by rmcgibbo** [comment]
Hi all!

I think this recipe is ready to be merged. I've fixed the issues mentioned in the comments, and built from the latest github release tarball.

**Comment by scopatz** [comment]
I think the name collision here is problematic.  Any alternative names for this package would be great

**Comment by rmcgibbo** [comment]
@scopatz: How about we change the conda-forge name to "desres-msys"? Of course we will be keeping the name of the Python module the same.

**Comment by scopatz** [comment]
Yeah, that works, the only thing that needs to change is the conda package name, not anything in Python at run time of course.

**Comment by rmcgibbo** [comment]
Alright, the package has been rebuilt with the new name.

File: recipes/msys/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{% set name = "desres-msys" %}
```
Comment (scopatz) [inline recipes/msys/meta.yaml]: Thanks!

File: recipes/msys/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+source:
+  - url: https://github.com/DEShawResearch/msys/archive/{{ version }}.tar.gz
+    sha1: {{ sha1 }}
+  - url: http://www.inchi-trust.org/download/105/INCHI-1-SRC.zip
+    sha1: e3872a46d58cb321a98f4fd4b93a989fb6920b9c
+    folder: external/inchi/INCHI-1-SRC
+  - url: https://sourceforge.net/projects/lpsolve/files/lpsolve/5.5.2.5/lp_solve_5.5.2.5_source.tar.gz
+    folder: external/lpsolve/lp_solve_5.5
+    sha1: 232ce8edb073c7c9b6d9100abf77038d989c45c8
```
Comment (scopatz) [inline recipes/msys/meta.yaml]: Isn't there an `lpsolve55` package already that you can link against?

File: recipes/msys/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/DEShawResearch/msys/archive/{{ version }}.tar.gz
+    sha1: {{ sha1 }}
+  - url: http://www.inchi-trust.org/download/105/INCHI-1-SRC.zip
+    sha1: e3872a46d58cb321a98f4fd4b93a989fb6920b9c
+    folder: external/inchi/INCHI-1-SRC
```
Comment (scopatz) [inline recipes/msys/meta.yaml]: Shouldn't this be its own package?

Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: Sure. Done.

File: recipes/msys/meta.yaml (hunk 4 above)
Comment (rmcgibbo) [reply recipes/msys/meta.yaml]: Yep, done

**Comment by rmcgibbo** [comment]
@skopatz: I think all your comments have been addressed now

**Comment by scopatz** [comment]
Thanks! Sorry, I missed this due to my name misspelling.

**Comment by rmcgibbo** [comment]
Sorry about that. I apologize for misspelling your name.

**Pull request merged**